jax/_src/api.py

```diff
@@ -3245,6 +3246,50 @@ def try_to_block(x):
       return x
   return jax.tree_util.tree_map(try_to_block, x)
 
+def pure_callback(callback: Callable[..., Any], result_shape_dtypes: Any,
+                  *args: Any, **kwargs: Any):
+  """Calls a pure Python callback function from staged out JAX programs.
+
+  ``pure_callback`` enables calling a Python function in JIT-ed JAX functions.
+  The input ``callback`` will be passed NumPy arrays in place of JAX arrays and
+  should also return NumPy arrays. Execution takes place on the CPU host.
```

Comment:
```suggestion
  should also return NumPy arrays. Execution takes place on CPU, like any Python+NumPy function.
```

Also, what happens if we use jnp? Undefined behavior? Possible hangs?